### PR TITLE
configurable failure policy

### PR DESCRIPTION
**Pull request by pperiyasamy**
provide a flexibility to configure failure policy for the web hook, potential fix for the issue #101

### CODE REVIEW

File: pkg/installer/installer.go [hunk 1]
```diff
@@ -127,11 +127,18 @@ func writeToFile(certificate, key []byte, certFilename, keyFilename string) erro
...
 }
 
-func createMutatingWebhookConfiguration(certificate []byte) error {
+func createMutatingWebhookConfiguration(certificate []byte, failurePolicyStr string) error {
 	configName := strings.Join([]string{prefix, "mutating-config"}, "-")
 	serviceName := strings.Join([]string{prefix, "service"}, "-")
 	removeMutatingWebhookIfExists(configName)
-	failurePolicy := arv1beta1.Ignore
+	var failurePolicy arv1beta1.FailurePolicyType
+	if failurePolicyStr == "Ignore" {
```
Comment (martinkennelly) [inline pkg/installer/installer.go:135]: can you do case insensitive comparison?

File: pkg/installer/installer.go [hunk 2]
```diff
@@ -127,11 +127,18 @@ func writeToFile(certificate, key []byte, certFilename, keyFilename string) erro
...
+		failurePolicy = arv1beta1.Ignore
+	} else if failurePolicyStr == "Fail" {
```
Comment (martinkennelly) [inline pkg/installer/installer.go:137]: can you do case insensitive comparison?

File: cmd/installer/main.go [hunk 3]
```diff
@@ -16,15 +16,17 @@ package main
...
 	"flag"
+
 	"github.com/golang/glog"
 	"github.com/k8snetworkplumbingwg/network-resources-injector/pkg/installer"
 )
 
 func main() {
 	namespace := flag.String("namespace", "kube-system", "Namespace in which all Kubernetes resources will be created.")
 	prefix := flag.String("name", "network-resources-injector", "Prefix added to the names of all created resources.")
+	failurePolicy := flag.String("failure-policy", "Ignore", "failure policy to handle unrecognized errors and timeout errors")
```
Comment (martinkennelly) [inline cmd/installer/main.go:27]: Can you extend the description to say 

> K8 admission controller failure policy to handle unrecognized errors and timeout errors

**Comment by martinkennelly** [comment]
@pperiyasamy I see this patch is depending/containing a commit from PR #102. 
@zshi-redhat I think failpolicy should be fail by default. I'd like to hear your opinion on this.
@pperiyasamy If so, we may need to update our deployment k8 [artifact ](https://github.com/k8snetworkplumbingwg/network-resources-injector/blob/master/deployments/server.yaml)as well but please wait until Zenghui has weighted in.

**Comment by zshi-redhat** [comment]
> @pperiyasamy I see this patch is depending/containing a commit from PR #102.
> @zshi-redhat I think failpolicy should be fail by default. I'd like to hear your opinion on this.
> @pperiyasamy If so, we may need to update our deployment k8 [artifact ](https://github.com/k8snetworkplumbingwg/network-resources-injector/blob/master/deployments/server.yaml)as well but please wait until Zenghui has weighted in.

@martinkennelly From [here](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy), it says admission V1 uses `Fail` as default `FailurePolicy`, admission V1beta uses `Ignore` as default `FailurePolicy`. 

Currently we are still using V1beta in NRI, maybe `Ignore` could be the default? We can switch the default to `Fail` once moving to admission V1. wdyt?

**Comment by martinkennelly** [comment]
> > @pperiyasamy I see this patch is depending/containing a commit from PR #102.
> > @zshi-redhat I think failpolicy should be fail by default. I'd like to hear your opinion on this.
> > @pperiyasamy If so, we may need to update our deployment k8 [artifact ](https://github.com/k8snetworkplumbingwg/network-resources-injector/blob/master/deployments/server.yaml)as well but please wait until Zenghui has weighted in.
> 
> @martinkennelly From [here](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy), it says admission V1 uses `Fail` as default `FailurePolicy`, admission V1beta uses `Ignore` as default `FailurePolicy`.
> 
> Currently we are still using V1beta in NRI, maybe `Ignore` could be the default? We can switch the default to `Fail` once moving to admission V1. wdyt?

Missed that. Thanks!
It seems reasonable to upgrade to V1 with #96 and get fail by default but I like this patch to make failure configurable. This is new behavior for NRI so we need to make it clear in release notes.

**Review by zshi-redhat — APPROVED** [review]

File: cmd/installer/main.go (hunk 3 above)
Comment (pperiyasamy) [reply cmd/installer/main.go]: done.

File: pkg/installer/installer.go (hunk 1 above)
Comment (pperiyasamy) [reply pkg/installer/installer.go]: done.

File: pkg/installer/installer.go (hunk 2 above)
Comment (pperiyasamy) [reply pkg/installer/installer.go]: done.

**Review by martinkennelly — APPROVED** [review]

**Pull request merged**